llvm/test/Transforms/AtomicExpand/AMDGPU/expand-atomic-i16.ll

```diff
@@ -1262,6 +1262,110 @@ define bfloat @test_atomicrmw_xchg_bf16_global_agent_align4(ptr addrspace(1) %pt
   ret bfloat %res
 }
 
+define i16 @test_atomicrmw_xchg_i16_buffer_fat_agent(ptr addrspace(7) %ptr, i16 %value) {
+; CHECK-LABEL: @test_atomicrmw_xchg_i16_buffer_fat_agent(
+; CHECK-NEXT:    [[ALIGNEDADDR:%.*]] = call ptr addrspace(7) @llvm.ptrmask.p7.i32(ptr addrspace(7) [[PTR:%.*]], i32 -4)
```

Comment:
Not sure this is really sensible. I'm surprised this is i32 and not i64, and since this is zero extended, this is probably nonsense

Reply:
Unsigned offsets are so problematic. I guess we could interpret this as a mask on the base address instead? But for that, we would want this to be the 64/48-bit mask and not the 32-bit indexing type

Reply:
global/scratch instructions are supposed to support signed offsets, but it was broken for a few generations.

None of this matters for LDS or those cases since they are normal pointers